qa/integration-tests/src/test/java/io/zeebe/broker/it/DataDeleteTest.java

```diff
@@ -0,0 +1,124 @@
+  private static final String SNAPSHOT_PERIOD = "30s";
+  private static final String PROCESS = "process";
+  private static final BpmnModelInstance WORKFLOW =
+      Bpmn.createExecutableProcess(PROCESS).startEvent("start").endEvent("end").done();
+
+  public EmbeddedBrokerRule brokerRule =
+      new EmbeddedBrokerRule(
+          brokerCfg -> {
+            final DataCfg data = brokerCfg.getData();
+            data.setMaxSnapshots(MAX_SNAPSHOTS);
+            data.setSnapshotPeriod(SNAPSHOT_PERIOD);
+            data.setDefaultLogSegmentSize("8k");
+            data.setIndexBlockSize("2K");
+
+            final List<ExporterCfg> exporters =
+                new ArrayList<ExporterCfg>() {
+                  {
+                    add(createExporter());
+                    add(createExporter());
+                  }
+                };
+
+            brokerCfg.setExporters(exporters);
+          });
+  public GrpcClientRule clientRule = new GrpcClientRule(brokerRule);
+
+  @Rule
+  public RecordingExporterTestWatcher recordingExporterTestWatcher =
+      new RecordingExporterTestWatcher();
+
+  @Rule public RuleChain ruleChain = RuleChain.outerRule(brokerRule).around(clientRule);
+  @Rule public Timeout timeout = new Timeout(60, TimeUnit.SECONDS);
+
+  @Test
+  public void shouldDeleteDataWithExporters() {
+    // given
+    final String rootPath = brokerRule.getBrokerCfg().getData().getDirectories().get(0);
+    final String snapshotDirPath = rootPath + "/partition-1/state/1_zb-stream-processor/snapshots";
+    final String segmentsDirPath = rootPath + "/partition-1/segments";
+    brokerRule.getClock().pinCurrentTime();
+
+    clientRule
+        .getClient()
+        .newDeployCommand()
+        .addWorkflowModel(WORKFLOW, "process.bpmn")
+        .send()
+        .join();
+
+    final File segmentsDir = new File(segmentsDirPath);
+    long lastWorkflowKey = -1;
+
+    while (segmentsDir.list().length <= 1) {
+      lastWorkflowKey =
+          clientRule
+              .getClient()
+              .newCreateInstanceCommand()
+              .bpmnProcessId(PROCESS)
+              .latestVersion()
+              .send()
+              .join()
+              .getWorkflowInstanceKey();
+    }
+
+    // when
+    final int segments = segmentsDir.list().length;
+    final long finalLastWorkflowKey = lastWorkflowKey;
+    TestUtil.waitUntil(
+        () -> TestExporter.records.stream().anyMatch(r -> r.getKey() == finalLastWorkflowKey));
+
+    brokerRule.getClock().addTime(Duration.ofSeconds(30));
+    final File snapshotsDir = new File(snapshotDirPath);
+    TestUtil.waitUntil(
+        () -> Arrays.stream(snapshotsDir.listFiles()).anyMatch(f -> !f.getName().equals("/tmp")));
+
+    // then
+    Assertions.assertThat(segmentsDir.list().length).isLessThan(segments);
+  }
+
+  private static ExporterCfg createExporter() {
+    final ExporterCfg exporterCfg = new ExporterCfg();
+    exporterCfg.setClassName(TestExporter.class.getName());
```

Comment:
We should use a different exporter. This one is defined in a different test module.